src/wasm/wasm-emscripten.cpp

```diff
@@ -170,9 +180,7 @@ std::string codeForConstAddr(Module& wasm,
                              int32_t address) {
   const char* str = stringAtAddr(wasm, segmentOffsets, address);
   if (!str) {
-    // If we can't find the segment corresponding with the address, then we
-    // omitted the segment and the address points to an empty string.
-    return escape("");
+    Fatal() << "unable to find data for ASM/EM_JS const at: " << address;
```

Comment:
What's this change about?

Reply:
I notices that if the segments offset are not calcualted the fallback of stringAtAddr is currently to return empty string which is really not useful.  

This would mean that emscripten gets passed the empty string for its EM_ASM constants..   I think its better to fail hard here.

I tests this with all the emscripten tests to make we don't hit this in any normal use cases.

Reply:
I'm somewhat worried by the comment I'm removing here though:  `// omitted the segment and the address points to an empty string.`

I don't see who a string could simultaneously be accessed for code and also `omitted`.    If the string was omitted it is a by definition not refernced.  If a string is referenced by omitted that would be bug and we want to fail as early as possible (IIUC).

Reply:
I agree that failing hard and fast sounds much better here 👍